docs/openapi/definitions/auth.yaml

```diff
@@ -0,0 +1,281 @@
+            Short-term authorization code used to be exchanged for access tokens
+          required: true
+          schema:
+            type: string
+        - name: state
+          in: query
+          description: Nonce value stored in state
+          required: true
+          schema:
+            type: string
+        - name: provider-nonce
+          in: cookie
+          description: Nonce value stored in cookie
+          required: true
+          schema:
+            type: string
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        200:
+          description: Message received from consent popup
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/AuthResponse'
+                  - $ref: '#/components/schemas/AuthError'
+              examples:
+                AuthResponse:
+                  summary: AuthResponse
+                  value: |
+                    {
+                      type: "auth-result",
+                      payload: {
+                          accessToken: "access-token",
+                          idToken: "id-token",
+                          expiresInSecond: 3600,
+                          scope: "email"
+                        }
+                    }
+                AuthError:
+                  summary: AuthError
+                  value: |
+                    {
+                      type: "auth-result",
+                      error: {
+                        name: "error",
+                        message: "message",
+                        stack: "stack"
+                      }
+                    }
+          headers:
+            Set-Cookie:
+              description: Refresh cookie
+              schema:
+                $ref: '#/components/schemas/Refresh-Cookie'
+  /refresh:
+    get:
+      tags:
+        - provider
+      summary: Handles refreshing of tokens due to reload or expiry
+      description:
+        If supported by the provider, the `/refresh` endpoint is responsible for
+        refreshing the tokens using a refresh-token stored in a cookie
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: optional
+          in: query
+          description: Prevent the popup from being displayed
+          schema:
+            type: string
+        - name: x-requested-with
+          in: header
+          description: X-Requested-With header preventing CSRF
+          required: true
+          schema:
+            type: string
+            example: 'x-requested-with: XMLHttpRequest'
+        - name: provider-refresh-token
```

Comment:
this is actually `google-refresh-token` for example, right? is it possible to match by regex or so?

Reply:
Yeah afaik, although come to think of it it's a bit of an implementation detail of how we handle refresh, we don't really need to enforce this method for all providers. i.e. it could be in our implementation docs but not the API Schema